lib/stanford-mods/date_parsing.rb

```diff
@@ -36,6 +37,7 @@ def self.year_str_valid?(year_str)
 
       def initialize(date_str)
         @orig_date_str = date_str
+        @orig_date_str.freeze
```

Comment:
If I'm not mistaken `#freeze` returns `self` so this may be more succinctly written as `@orig_date_str = date_str.freeze`

Up to you.